src/components/pill/pill.style.js

```diff
@@ -60,211 +42,206 @@ const PillStyle = styled.span`
+            padding: 0 41px 0 19px;
+            border-radius: 15px;
 
-          button {
-            width: 32px;
-            padding: 0;
-            border-radius: 0 12px 12px 0;
-            line-height: 18px;
+            button {
+              width: 32px;
+              padding: 0;
+              border-radius: 0 12px 12px 0;
+              line-height: 18px;
 
-            ${addStyleToPillIcon('3px 9px 5px 9px', '13px', '1px')}
-          }
+              ${addStyleToPillIcon('3px 9px 5px 9px', '13px', '1px')}
+            }
+          `}
         `}
-      `}
 
-      ${!isClassic(theme, colorVariant) && !isDeletable && css`
-        ${size === 'S' && css`
-          padding: 0 7px;
+        ${!isDeletable && css`
+          ${size === 'S' && css`
+            padding: 0 7px;
 
-          button {
-            padding: 0;
-          }
-        `}
+            button {
+              padding: 0;
+            }
+          `}
 
-        ${size === 'M' && css`
-          padding: 0 11px;
-          border-radius: 12px;
+          ${size === 'M' && css`
+            padding: 0 11px;
+            border-radius: 12px;
 
-          button {
-            width: 24px;
-            padding: 0;
-            border-radius: 0 8px 8px 0;
-          }
-        `}
+            button {
+              width: 24px;
+              padding: 0;
+              border-radius: 0 8px 8px 0;
+            }
+          `}
 
-        ${size === 'L' && css`
-          padding: 0 15px;
-          border-radius: 13px;
+          ${size === 'L' && css`
+            padding: 0 15px;
+            border-radius: 13px;
 
-          button {
-            width: 28px;
-            padding: 0;
-            border-radius: 0 10px 10px 0;
-          }
-        `}
+            button {
+              width: 28px;
+              padding: 0;
+              border-radius: 0 10px 10px 0;
+            }
+          `}
 
-        ${size === 'XL' && css`
-          padding: 0 19px;
-          border-radius: 15px;
+          ${size === 'XL' && css`
+            padding: 0 19px;
+            border-radius: 15px;
 
-          button {
-            width: 32px;
-            padding: 0;
-            border-radius: 0 12px 12px 0;
-          }
+            button {
+              width: 32px;
+              padding: 0;
+              border-radius: 0 12px 12px 0;
+            }
+          `}
         `}
       `}
 
-      ${isClassic(theme, colorVariant) && classicThemeForPill(colorVariant, inFill, isDeletable, size)}
+      ${isClassic(theme) && classicThemeForPill(colorVariant, inFill, isDeletable, size)}
     `;
   }
 }
 `;
 
-function isClassic(theme, colorVariant) {
-  // handles incorrect default activeTheme in StoryBook AppWrapper
-  if (OptionsHelper.pillColors.includes(colorVariant)) {
-    return false;
-  }
-  return isClassicTheme(theme);
+function hasCarbonColorVariant(colorVariant) {
+  return [...OptionsHelper.pillColors, 'warning'].includes(colorVariant);
```

Comment:
the whole component needs a solid refactor. In this task I have only refactored it to not rely on specific themes. I have opted to not dig deeper into that as it may break something else.

Reply:
OK thanks :thumbsup: